Update the Helm chart README.md with Kubernetes support changes and upgrade changes (#7327)
strimzi/strimzi-kafka-operator: packaging/helm-charts/helm3/strimzi-kafka-operator/README.md

[Kubernetes](https://kubernetes.io/) or [OpenShift](https://www.openshift.com/) in various deployment configurations.
See our [website](https://strimzi.io) for more details about the project.

**!!! IMPORTANT !!!**

> Upgrading to Strimzi 0.32 and newer directly from Strimzi 0.22 and earlier is not possible anymore.

Reviewer: ```suggestion
Upgrading to Strimzi 0.32 and newer directly from Strimzi 0.22 and earlier is no longer possible.
```
